apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c

```diff
@@ -74,241 +63,754 @@ static void clean_all_data(int fd, char *buffer)
+	const char *ptr_flag[] = { "PCM_IN", "PCM_OUT", "PCM_MMAP", "PCM_NOIRQ", "PCM_MONOTONIC" };
+	count = sizeof(flag_arr) / sizeof(flag_arr[0]);
+
+	for (index = 0; index < count; index++) {
+		g_pcm = NULL;
+		g_pcm = pcm_open(0, 0, flag_arr[index], NULL);
+		if (g_pcm == NULL || pcm_get_file_descriptor(g_pcm) <= 0) {
+			printf("\nitc_audio_pcm_open_p_with_different_flag: pcm_open FAIL for flag : %s\n", ptr_flag[index]);
+			check = false;
+			continue;
+		}
+		ret = pcm_close(g_pcm);
+		if (ret != 0) {
+			printf("\nitc_audio_pcm_open_p_with_different_flag: pcm_close FAIL for flag : %s\n", ptr_flag[index]);
+			check = false;
+		}
+	}
+
+	TC_ASSERT_EQ("itc_audio_pcm_open_p_with_different_flag", check, true);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_open_p_reopen
+* @brief            open and initlaize pcm structure
+* @scenario         Open pcm and check its validation
+* @apicovered       pcm_open, pcm_close
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_audio_pcm_open_p_reopen(void)
+{
+	struct pcm *ptr_pcm = NULL;
+	g_pcm = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+	// re-open pcm stream
+	ptr_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_GT_CLEANUP("pcm_open", pcm_get_file_descriptor(ptr_pcm), 0, pcm_close(g_pcm));
+
+	TC_ASSERT_EQ("pcm_close", pcm_close(ptr_pcm), 0);
+	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_open_n
+* @brief            open and initlaize pcm structure with invalid values
+* @scenario         Open pcm with invalid device & card value and check its validation
+* @apicovered       pcm_open
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_audio_pcm_open_n(void)
+{
+	struct pcm *ptr_pcm = NULL;
+
+	ptr_pcm = pcm_open(999, 999, PCM_IN, NULL);
+	TC_ASSERT_LEQ_CLEANUP("pcm_open", pcm_get_file_descriptor(ptr_pcm), 0, pcm_close(ptr_pcm));
+
+	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_close_n
+* @brief            close opened pcm
+* @scenario         close opened pcm with null pcm
+* @apicovered       pcm_close
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_audio_pcm_close_n(void)
+{
+	TC_ASSERT_NEQ("pcm_close", pcm_close(NULL), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_is_ready_p
 * @brief            open and check pcm is ready or not
 * @scenario         Open pcm and check its state
 * @apicovered       pcm_is_ready
 * @precondition     pcm_open should be called
 * @postcondition    NA
 */
-static void itc_audio_pcm_is_ready_tc_p(void)
+static void itc_audio_pcm_is_ready_p(void)
+{
+	g_pcm = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
```

Comment:
Cleanup is needed here  to close g_pcm.

Reply:
There is no need to close g_pcm, as it is not yet open. This Assert checks whether pcm_open API is actually open or not, if it is not opened, then it cannot be closed